common/components/UI/ModalWindow/index.tsx

```diff
@@ -30,20 +30,18 @@ const Modal: React.FC<Props> = ({
   className,
   maskClickIgnore,
   style,
-  // footer,
   title,
   okButtonProps,
   open = true,
-  preview = false,
+  preview,
+  cancelButtonProps,
 }) => {
-
   const handleCancel = () => {
-    const isSingleTabAndViewMode = changesForm() && open && preview;
+    const isSingleTabAndViewMode = changesForm() && open && preview
 
-    if(isSingleTabAndViewMode){
-      onCancel();
-    }
-    else {
+    if (isSingleTabAndViewMode || preview) {
+      onCancel()
+    } else {
```

Comment:
Let's make this conditional block as:

`edit` - pop-confirm if form was changed
`create` - pop-confirm if form was changes
`preview` - close without pop-confirm and `onCancel` callback (we just closing modal, we don't undo any changes, so... we don't need to receive `onCancel` event)

To track form changes use `Form#onChange`